src/site/stages/build/drupal/page.js

```diff
@@ -49,7 +49,25 @@ function paginatePages(page, files, field, layout, ariaLabel, perPage) {
     ariaLabel = ` of ${ariaLabel}`;
   }
 
-  const pagedEntities = _.chunk(page[field].entities, perPage);
+  let chunker;
+  switch (page.entityBundle) {
+    case 'event_listing':
+      chunker = page.allEventTeasers.entities;
+      break;
+    case 'story_listing':
+      chunker = page.allNewsStoryTeasers.entities;
+      break;
+    case 'leadership_listing':
+      chunker = page.fieldLeadership;
+      break;
+    case 'press_releases_listing':
+      chunker = page.allPressReleaseTeasers.entities;
+      break;
+    default:
+      chunker = page[field].entities;
+  }
+  const pagedEntities = _.chunk(chunker, perPage);
```

Comment:
I think this could be simplified a bit.  The goal seems to be changing what `field` is based on `page.entityBundle`, but since `field` is a param it probably isn't a good idea to actually reassign it.  What about something like this?


```js
  /* eslint-disable camelcase */
  // Map the page.entityBundle value to a property value we want for page
  const bundleToField = {
    event_listing: 'allEventTeasers',
    story_listing: 'allNewsStoryTeasers',
    leadership_listing: 'fieldLeadership',
    press_releases_listing: 'allPressReleaseTeasers',
  };
  /* eslint-enable camelcase */
  const pageField = _.get(bundleToField, page.entityBundle, field);
  const pagedEntities = _.chunk(page[pageField].entities, perPage);
```
Or if you renamed `field` to `defaultField`, maybe something like this:

```js
const field = bundleToField[page.entityBundle]
const pagedEntities = _.chunk(page[field || defaultField], perPage)
```

I haven't tested this - it's just an idea since the switch statement seems a bit overkill.

Reply:
will do